backend/deepchecks_monitoring/bgtasks/model_data_ingestion_alerter.py

```diff
@@ -0,0 +1,155 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+import pendulum as pdl
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import insert
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm import selectinload
+
+from deepchecks_monitoring.public_models.organization import Organization
+from deepchecks_monitoring.public_models.task import UNIQUE_NAME_TASK_CONSTRAINT, BackgroundWorker, Task
+from deepchecks_monitoring.resources import ResourcesProvider
+from deepchecks_monitoring.schema_models import Model
+from deepchecks_monitoring.schema_models.column_type import SAMPLE_ID_COL, SAMPLE_LABEL_COL, SAMPLE_TS_COL
+from deepchecks_monitoring.schema_models.data_ingestion_alert import DataIngestionAlert
+from deepchecks_monitoring.schema_models.monitor import Frequency, as_pendulum_datetime
+from deepchecks_monitoring.utils import database
+
+__all__ = ["ModelDataIngestionAlerter"]
+
+
+QUEUE_NAME = "model_data_ingestion alerter"
+DELAY = 60
+
+
+class ModelDataIngestionAlerter(BackgroundWorker):
+    """Worker to delete kafka topics when they are no longer in use."""
+
+    @classmethod
+    def queue_name(cls) -> str:
+        return QUEUE_NAME
+
+    @classmethod
+    def delay_seconds(cls) -> int:
+        return DELAY
+
+    async def run(self, task: "Task",
+                  session: AsyncSession,  # pylint: disable=unused-argument
+                  resources_provider: ResourcesProvider):
+        model_id = task.params["model_id"]
+        org_id = task.params["organization_id"]
+        end_time = task.params["end_time"]
+        start_time = task.params["start_time"]
+
+        organization_schema = (await session.execute(
+            sa.select(Organization.schema_name).where(Organization.id == org_id)
+        )).scalar_one_or_none()
+
+        # If organization was removed doing nothing
+        if organization_schema is None:
+            return
+
+        await database.attach_schema_switcher_listener(
+            session=session,
+            schema_search_path=[organization_schema, "public"]
+        )
+
+        model: Model = (
+            await session.execute(sa.select(Model).where(Model.id == model_id).options(selectinload(Model.versions)))
+        ).scalars().first()
+        freq: Frequency = model.data_ingestion_alert_frequency
```

Comment:
model might be deleted between task creation to running, missing `if model is None`